design/proposals/load_policy/load_policy_validation.md

```diff
@@ -0,0 +1,257 @@
+# Solution Design - Load policy validation
+
+## Table of Contents
+* [Glossary](#glossary)
+* [Useful Links](#useful-links)
+* [Issue Description](#issue-description)
+* [Solution](#solution)
+   + [Solution - scope](#solution---scope)
+   + [User Interface](#user-interface)
+* [Design](#design)
+   + [Flow Diagrams](#flow-diagrams)
+   + [Sequence Diagrams](#sequence-diagrams)
+      - [Sequence Diagrams - additional info](#sequence-diagrams---additional-info)
+* [Performance](#performance)
+* [Backwards Compatibility](#backwards-compatibility)
+* [Affected Components](#affected-components)
+* [Test Plan](#test-plan)
+   + [Out of Scope](#out-of-scope)
+   + [Prerequisites](#prerequisites)
+* [Documentation](#documentation)
+* [Security](#security)
+* [Infrastructure](#infrastructure)
+* [Audit](#audit)
+* [Open Questions](#open-questions)
+* [Delivery Plan](#delivery-plan)
+   + [Total 1st milestone EE 45](#total-1st-milestone-ee-45)
+      - [Ramp-up EE 2](#ramp-up-ee-2)
+      - [Designs EE 2](#designs-ee-2)
+      - [Implementations EE 19](#implementations-ee-19)
+      - [Testing EE 13](#testing-ee-13)
+      - [Security EE 5](#security-ee-5)
+      - [Docs EE 3](#docs-ee-3)
+      - [Demo EE 1](#demo-ee-1)
+* [Definition of Done](#definition-of-done)
+* [Solution Review](#solution-review)
+
+## Useful Links
+- [Feature doc](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-8566)
+- [UX research page](https://ca-il-confluence.il.cyber-ark.com.mcas.ms/display/rndp/JWT+Authenticator+-+load+policy+validation)
+- [JWT authenticator stage 1.1 feature doc](https://ca-il-confluence.il.cyber-ark.com.mcas.ms/display/rndp/JWT+Authenticator+-+re-scope#JWTAuthenticatorrescope-Stage1.1-SimplegenericJWTauthenticatorwithtokenschema)
+
+
+## Issue Description
+In misconfiguration errors, for example:
+
+for hosts without JWT **mandatory** annotation,
+
+We are failing at JWT authentication request with 401 error and with no additional data in the response, which is a bad UX.
+By adding the load policy validation it is improving the UX, and will help the customer to create a valid policies.
+
+
+## Solution
+For better UX we want to perform policy validation steps in order to identify configuration errors as early as possible. 
+
+### Solution - scope
+This solution is creating a generic infrastructure for all policy entities validation.
+And focusing on Host with JWT authenticator annotations as first use case to implement on this new suggested infrastructure.
+
+### UX
+#### Load JWT policy without mandatory JWT annotation
```

Comment:
same question - what about with mandatory fields?

Reply:
we should include mapping in here

Reply:
As i see it this is an high level design and infrastructure to validate all current and future use cases of JWT so i didn't want to focus on specific use cases